### PR TITLE
Separate modules

### PR DESCRIPTION
**Pull request by nickhsmith**
Separate modules using config file options.

update submodule to have workDir attributes
update cli to correctly update drop from base dir
update Snakefile handling and shared input/output generation
method to ignore modules in wBuild by changing the filename to an ignorable path

### CODE REVIEW

File: drop/template/Scripts/AberrantExpression/Overview.R [hunk 1]
```diff
@@ -22,12 +23,11 @@
 #'    code_download: TRUE
 #'---
 
-#+ echo=F
-saveRDS(snakemake, snakemake@log$snakemake)
+#+ include=FALSE
+knitr::opts_chunk$set(eval = snakemake@params$run)
```
Comment (mumichae) [inline drop/template/Scripts/AberrantExpression/Overview.R]: Clean up

File: drop/template/Scripts/AberrantSplicing/Overview.R [hunk 2]
```diff
@@ -21,13 +22,14 @@
 #'    code_download: TRUE
 #'---
 
-#+ echo=F
+
+#+ include=FALSE
+knitr::opts_chunk$set(eval = snakemake@params$run)
+
+#+ echo=TRUE, eval = TRUE
```
Comment (mumichae) [inline drop/template/Scripts/AberrantSplicing/Overview.R]: clean up

File: drop/template/Snakefile [hunk 3]
```diff
@@ -10,24 +11,67 @@ drop.checkDropVersion(projectDir)
...
 config = cfg.config_dict # legacy
 
+def find_snakefile(module):
+    if os.path.isdir(module.getWorkdir()):
+        return module.getWorkdir() + "/Snakefile"
+    else:
+        return module.getWorkdir(hide_dir = True) + "/Snakefile"
 
+
+def moduleInputOutput(index_name, work_dir,module):
```
Comment (mumichae) [inline drop/template/Snakefile]: rename to getModuleIndexFiles

File: drop/template/Snakefile [hunk 4]
```diff
@@ -10,24 +11,67 @@ drop.checkDropVersion(projectDir)
...
+            index_name= index_name
+        )
+        return index_input,graph_file,index_output
+    else:
+        #dummy return of hidden pipeline directory which already exists
+        return module.getWorkdir(hide_dir = True),module.getWorkdir(hide_dir = True),module.getWorkdir(hide_dir = True)
+
+def renameLocalDir(module):
+    work_dir = module.getWorkdir(str_=False)
+    work_dir_prefix = str(work_dir).rsplit("Scripts",1)[0] + "Scripts/"
```
Comment (mumichae) [inline drop/template/Snakefile]: projectDir + /'Scripts/'

File: drop/template/Snakefile (hunk 3 above)
Comment (mumichae) [reply drop/template/Snakefile]: Move to Config class

File: drop/template/Snakefile [hunk 5]
```diff
@@ -10,24 +11,67 @@ drop.checkDropVersion(projectDir)
...
+        index_input, index_output, graph_file, _ = createIndexRule(
+            scriptsPath=str(work_dir),
+            index_name= index_name
+        )
+        return index_input,graph_file,index_output
+    else:
+        #dummy return of hidden pipeline directory which already exists
+        return module.getWorkdir(hide_dir = True),module.getWorkdir(hide_dir = True),module.getWorkdir(hide_dir = True)
+
+def renameLocalDir(module):
```
Comment (mumichae) [inline drop/template/Snakefile]: move to Submodule

File: drop/template/config.yaml [hunk 6]
```diff
@@ -13,18 +13,19 @@ geneAnnotation:
 
 genomeAssembly: hg19  # either hg19/hs37d5 or hg38/GRCh38
 exportCounts:
-    # specify which gene annotations to include and which 
+    # specify which gene annotations to include and which
     # groups to exclude when exporting counts
     geneAnnotations:
       - v29
-    excludeGroups:
+    excludeGroupsu
```
Comment (mumichae) [inline drop/template/config.yaml]: fix

File: tests/common.py [hunk 7]
```diff
@@ -7,6 +7,7 @@
 
 
 def run(cmd, dir_path, report_stdout=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE, **kwargs):
+    print(cmd)
```
Comment (mumichae) [inline tests/common.py]: remove

File: tests/pipeline/test_AE.py [hunk 8]
```diff
@@ -3,6 +3,16 @@
 
 class Test_AE_Pipeline:
 
+    def test_pipeline_no_run(self,demo_dir):
+        LOGGER.info("run aberrantExpression pipeline with \'run: false\'")
+        run("awk -v n=1 \'/run: true/ { if (++count == n) sub(/run: true/, \"run: false\"); } 1\' \
```
Comment (mumichae) [inline tests/pipeline/test_AE.py:10]: comment on what awk does here

File: drop/template/Snakefile (hunk 3 above)
Comment (nickhsmith) [reply drop/template/Snakefile]: moved to submodule class. Because this happens at the sub module level

File: drop/template/Snakefile [hunk 9]
```diff
@@ -10,24 +11,24 @@ drop.checkDropVersion(projectDir)
...
+include: cfg.AE.getSnakefile()
+include: cfg.AS.getSnakefile()
+include: cfg.MAE.getSnakefile()
 include: drop.utils.getWBuildSnakefile()
-include: cfg.AE.getWorkdir() + "/Snakefile"
-include: cfg.AS.getWorkdir() + "/Snakefile"
-include: cfg.MAE.getWorkdir() + "/Snakefile"
 
 
 # don't allow the dataset wildcard to contain '--' as this is a delimter
```
Comment (nickhsmith) [inline drop/template/Snakefile]: delimiter

**Review by vyepez88 — APPROVED** [review]

**Pull request merged**